apps/homescreen/js/pages.js

```diff
@@ -219,9 +219,10 @@
       // Remove child immediately, datastore operations can be quite slow
       this.removeCard(this.selectedCard);
 
+      var revId = this.pagesStore.datastore.revisionId;
       this.pagesStore.get(id).then(entry => {
         entry.data.pinned = false;
-        this.pagesStore.datastore.put(entry.data, id).then(() => {},
+        this.pagesStore.datastore.put(entry.data, id, revId).then(() => {},
```

Comment:
Perhaps rather than showing a warning, it could just retry in X time (say 10 seconds) up to Y times (say 6 times)? It seems unlikely that it would fail over the course of a minute I would suppose, but if it did, perhaps we could show the warning then and add the pinned site back to the container? That last bit could be done by instead of immediately removing, setting display:none on the item and running .synchronise on the container - then if it fails, you can remove the display:none and run synchronise again to show it, and actually remove it on success.